source/extensions/filters/common/expr/custom_cel/example/example_custom_cel_vocabulary.cc

```diff
@@ -0,0 +1,113 @@
+#include "source/extensions/filters/common/expr/custom_cel/example/example_custom_cel_vocabulary.h"
+
+#include "envoy/extensions/expr/custom_cel_vocabulary/example/v3/config.pb.h"
+#include "envoy/extensions/expr/custom_cel_vocabulary/example/v3/config.pb.validate.h"
+#include "envoy/registry/registry.h"
+
+#include "source/extensions/filters/common/expr/custom_cel/custom_cel_vocabulary.h"
+#include "source/extensions/filters/common/expr/custom_cel/example/custom_cel_functions.h"
+#include "source/extensions/filters/common/expr/custom_cel/example/custom_cel_variables.h"
+
+#include "eval/public/activation.h"
+#include "eval/public/cel_function_adapter.h"
+#include "eval/public/cel_value.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Filters {
+namespace Common {
+namespace Expr {
+namespace Custom_Cel {
+namespace Example {
+
+void ThrowException(absl::string_view function_name, absl::Status status);
+
+void ExampleCustomCelVocabulary::fillActivation(Activation* activation, Protobuf::Arena& arena,
+                                                const StreamInfo::StreamInfo& info,
+                                                const Http::RequestHeaderMap* request_headers,
+                                                const Http::ResponseHeaderMap* response_headers,
+                                                const Http::ResponseTrailerMap* response_trailers) {
+  setRequestHeaders(request_headers);
+  setResponseHeaders(response_headers);
+  setResponseTrailers(response_trailers);
+  // variables
+  activation->InsertValueProducer(
+      CustomCelVariablesSetName,
+      std::make_unique<CustomCelVariablesWrapper>(arena, info, request_headers, response_headers,
+                                                  response_trailers));
+  // Lazily evaluated functions only
+  absl::Status status;
+  status =
+      activation->InsertFunction(std::make_unique<getDoubleCelFunction>(LazyEvalFuncNameGetDouble));
+  if (!status.ok()) {
+    ThrowException(LazyEvalFuncNameGetDouble, status);
+  }
+  status = activation->InsertFunction(
+      std::make_unique<getProductCelFunction>(LazyEvalFuncNameGetProduct));
+  if (!status.ok()) {
+    ThrowException(LazyEvalFuncNameGetProduct, status);
+  }
+  status = activation->InsertFunction(std::make_unique<get99CelFunction>(LazyEvalFuncNameGet99));
+  if (!status.ok()) {
+    ThrowException(LazyEvalFuncNameGet99, status);
+  }
+}
+
+void ExampleCustomCelVocabulary::registerFunctions(CelFunctionRegistry* registry) const {
```

Comment:
I guess maybe due to lack of docs on the API, but it's kind of hard to see whether this is exception safe. It's exception safe if this is only invoked during config load and when building the factory, but not when using the factory. Looking below, I think this is the case, but comments would help.

Reply:
I think it's not exception safe as is, and I'll remove the exceptions.